crates/bevy_hierarchy/src/child_builder.rs

```diff
@@ -499,13 +494,8 @@ impl<'w> WorldChildBuilder<'w> {
     pub fn spawn_empty(&mut self) -> EntityWorldMut<'_> {
         let entity = self.world.spawn(Parent(self.parent)).id();
         push_child_unchecked(self.world, self.parent, entity);
-        push_events(
-            self.world,
-            [HierarchyEvent::ChildAdded {
-                child: entity,
-                parent: self.parent,
-            }],
-        );
+        self.world
```

Comment:
This is an example of somewhere the observer will be called much sooner that an `EventReader<HierarchyEvent>` could have responded. It will see the `Entity`  just as it has been spawned, before any components could be added. Open Question: Is this desirable?